### PR TITLE
CRM-19876 add test to try to replicate token issue

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
* [CRM-19876: Civi won't send simple mails to all email recpients if some hold multiple addresses](https://issues.civicrm.org/jira/browse/CRM-19876)

### CODE REVIEW

File: CRM/Contact/Form/Task/EmailCommon.php [hunk 1]
```diff
@@ -74,18 +74,14 @@ public static function preProcessFromAddress(&$form, $bounce = TRUE) {
...
-    $contactID = $session->get('userID');
+    $contactID = CRM_Core_Session::singleton()->getLoggedInContactID();
+    $fromDisplayName  = CRM_Core_Session::singleton()->getLoggedInContactDisplayName();
 
     $form->_contactIds = array($contactID);
     $contactEmails = CRM_Core_BAO_Email::allEmails($contactID);
 
     $form->_onHold = array();
 
-    $fromDisplayName = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Contact',
```
Comment (eileenmcnaughton) [inline CRM/Contact/Form/Task/EmailCommon.php]: moved up a couple of lines

File: CRM/Contact/Form/Task/EmailCommon.php [hunk 2]
```diff
@@ -463,27 +472,6 @@ public static function postProcess(&$form) {
       $subject = "[case #$hash] $subject";
     }
 
-    // process message template
```
Comment (eileenmcnaughton) [inline CRM/Contact/Form/Task/EmailCommon.php]: extracted to own function

File: CRM/Utils/Token.php [hunk 3]
```diff
@@ -1184,11 +1184,7 @@ public static function getTokenDetails(
     $className = NULL,
     $jobID = NULL
   ) {
-    if (empty($contactIDs)) {
```
Comment (eileenmcnaughton) [inline CRM/Utils/Token.php]: how many years do we need to leave in debug?

**Review by seamuslee001 — APPROVED** [review]
This looks good haven't tested it locally but seems to make sense to me and looks to be clean up and tests should cover this

**Comment by eileenmcnaughton** [comment]
adding merge on pass per @seamuslee001 review

**Comment by jitendrapurohit** [comment]
@eileenmcnaughton seems this was to check https://issues.civicrm.org/jira/browse/CRM-19876. As the added test `testGetTokenDetailsMultipleEmails()` passes correctly, do we need two additional PRs for the same ticket - https://github.com/civicrm/civicrm-core/pull/10258 & https://github.com/civicrm/civicrm-core/pull/9676?

**Comment by eileenmcnaughton** [comment]
I'm not sure this really addressed the issue - we should check status of the bug with @lcdservices since I am pretty sure he took an interest.

**Pull request merged**